opus/application/static_media/js/browse.js

```diff
@@ -548,34 +521,102 @@ var o_browse = {
+                        if (!o_browse.isLastRowOfObservations(tab, viewNamespace, obsNum)) {
+                            detailOpusId = $("#galleryView").find(".op-next").data("id");
+                            o_browse.removeEditMetadataDetails();
+                            o_browse.loadPageIfNeeded("next", detailOpusId);
+                        } else {
+                            return false;
+                        }
                         break;
                     case 37:  // prev
+                        obsNum--;
                         detailOpusId = $("#galleryView").find(".op-prev").data("id");
                         o_browse.removeEditMetadataDetails();
                         o_browse.loadPageIfNeeded("prev", detailOpusId);
                         break;
+                    case 38:  // up
+                        offset = (o_browse.isGalleryView() ? viewNamespace.galleryBoundingRect.x : 1);
+                        if (obsNum > offset) {
+                            obsNum -= offset;
+                            detailOpusId = (o_browse.isGalleryView() ? $(tab).find(`.op-thumbnail-container[data-obs='${obsNum}']`).data("id") : $(tab).find(`tr[data-obs='${obsNum}']`).data("id"));
+                            o_browse.removeEditMetadataDetails();
+                            o_browse.loadPageIfNeeded("prev", detailOpusId);
+                        }
+                        break;
+                    case 40:  // down
+                        offset = (o_browse.isGalleryView() ? viewNamespace.galleryBoundingRect.x : 1);
+                        if (obsNum + offset <= viewNamespace.totalObsCount) {
+                            obsNum += offset;
+                            // if this new observation is in the last row of screen, don't allow the user to arrow there
+                            if (!o_browse.isLastRowOfObservations(tab, viewNamespace, obsNum)) {
+                                detailOpusId = (o_browse.isGalleryView() ? $(tab).find(`.op-thumbnail-container[data-obs='${obsNum}']`).data("id") : $(tab).find(`tr[data-obs='${obsNum}']`).data("id"));
+                                o_browse.removeEditMetadataDetails();
+                                o_browse.loadPageIfNeeded("next", detailOpusId);
+                            } else {
+                                return false;
+                            }
+                        }
+                        break;
                 }
                 if (detailOpusId && !$("#galleryViewContents").hasClass("op-disabled")) {
-                    o_browse.updateGalleryView(detailOpusId);
+                    o_browse.updateGalleryView(detailOpusId, obsNum);
+                    return false;
                 }
             }
             // don't return false here or it will snatch all the user input!
         });
     }, // end browse behaviors
 
+    isLastRowOfObservations: function(tab, viewNamespace, obsNum) {
```

Comment:
I'm confused by this function. The point is not to check for the last row on the current screen. The point is to check for the last row in the cached data regardless of whether it's displayed or not. This appears to be doing the former.